modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ddl/DdlCommandHandler.java

```diff
@@ -0,0 +1,281 @@
+
+        tableManager.alterTable(fullName, chng -> chng.changeIndices(idxes -> {
+            if (idxes.get(cmd.indexName()) == null) {
+                if (!cmd.ifExist()) {
+                    throw new IndexAlreadyExistsException(cmd.indexName());
+                }
+            }
+
+            idxes.delete(cmd.indexName());
+        }));
+    }
+
+    /**
+     * Adds a column according to the column definition.
+     *
+     * @param fullName Table with schema name.
+     * @param colsDef  Columns defenitions.
+     * @param colNotExist Flag indicates exceptionally behavior in case of already existing column.
+     */
+    private void addColumnInternal(String fullName, Set<ColumnDefinition> colsDef, boolean colNotExist) {
+        tableManager.alterTable(
+                fullName,
+                chng -> chng.changeColumns(cols -> {
+                    Map<String, String> colNamesToOrders = columnOrdersToNames(chng.columns());
+
+                    Set<ColumnDefinition> colsDef0;
+
+                    if (!colNotExist) {
+                        colsDef.stream().filter(k -> colNamesToOrders.containsKey(k.name())).findAny()
+                                .ifPresent(c -> {
+                                    throw new ColumnAlreadyExistsException(c.name());
+                                });
+
+                        colsDef0 = colsDef;
+                    } else {
+                        colsDef0 = colsDef.stream().filter(k -> !colNamesToOrders.containsKey(k.name())).collect(Collectors.toSet());
+                    }
+
+                    for (ColumnDefinition colDef : colsDef0) {
+                        cols.create(colDef.name(), colChg -> convert(colDef, colChg));
+                    }
+                }));
+    }
+
+    /**
+     * Adds a column according to the column definition.
+     *
+     * @param fullName Table with schema name.
+     * @param colNames  Columns defenitions.
+     * @param colNotExist Flag indicates exceptionally behavior in case of already existing column.
+     */
+    private void dropColumnInternal(String fullName, Set<String> colNames, boolean colExist) {
+        tableManager.alterTable(
+                fullName,
+                chng -> chng.changeColumns(cols -> {
+                    PrimaryKeyView priKey = chng.primaryKey();
+
+                    Map<String, String> colNamesToOrders = columnOrdersToNames(chng.columns());
+
+                    Set<String> colNames0 = new HashSet<>();
+
+                    for (String colName : colNames) {
+                        if (!colNamesToOrders.containsKey(colName)) {
+                            if (!colExist) {
+                                throw new ColumnNotFoundException(colName);
+                            }
+                        } else {
+                            colNames0.add(colName);
+                        }
+
+                        for (String priColName : priKey.columns()) {
```

Comment:
it would be better to put key columns in a set

Reply:
sorry i don`t understand here..

Reply:
Seems Konstantin means,
```
var primaryCols = new HashSet(priKey.columns());
....
for (String colName : colNames) {
    ....
    if (primaryCols.contains(colName)) 
```

Reply:
got it!